stacks-signer/src/signer.rs

```diff
@@ -1320,14 +1336,61 @@ impl Signer {
     fn save_signer_state_in_stackerdb(
         &mut self,
         encrypted_state: Vec<u8>,
-    ) -> Result<(), PersistenceError> {
+    ) -> Result<(), storage::PersistenceError> {
         let message = SignerMessage::EncryptedSignerState(encrypted_state);
 
         self.stackerdb.send_message_with_retry(message)?;
 
         Ok(())
     }
 
+    /// Load the saved signer state for the current dkg round
+    pub fn load_saved_state(&mut self) -> Result<(), storage::PersistenceError> {
+        let Some(aggregate_key) = self.approved_aggregate_public_key else {
```

Comment:
This check leads to surprising behavior.  Should this method, as named, _unconditionally_ load the state?

Reply:
Good point. Perhaps it's better to take the aggregate key as input, thus moving the condition to the calling code instead (unless the caller has the aggregate key already and not as an option). I'll change that.